lib/improver/tests/utilities/test_save.py

```diff
@@ -70,13 +84,44 @@ def test_basic(self):
         save_netcdf(self.cube, self.filepath)
         self.assertTrue(os.path.exists(self.filepath))
 
-    def test_saved_cube(self):
+    def test_cube_list(self):
+        """ Test functionality for saving iris.cube.CubeList """
+        cube_list = ([self.cube, self.cube])
+        save_netcdf(cube_list, self.filepath)
+        read_cubes = iris.load(self.filepath)
+        self.assertTrue(isinstance(read_cubes, iris.cube.CubeList))
+        self.assertEqual(len(read_cubes), 2)
+
+    def test_cube_data(self):
         """ Test valid cube can be read from saved file """
         save_netcdf(self.cube, self.filepath)
-        cube = iris.load_cube(self.filepath)
+        cube = load_cube(self.filepath)
         self.assertTrue(isinstance(cube, iris.cube.Cube))
         self.assertTrue(np.array_equal(cube.data, self.cube.data))
 
+    def test_cube_dimensions(self):
+        """ Test cube dimension coordinates are preserved """
+        save_netcdf(self.cube, self.filepath)
+        cube = load_cube(self.filepath)
+        coord_names = [coord.name() for coord in cube.coords(dim_coords=True)]
+        reference_names = [coord.name()
+                           for coord in self.cube.coords(dim_coords=True)]
+        self.assertItemsEqual(coord_names, reference_names)
+
+    def test_cf_global_attributes(self):
```

Comment:
I can't work out how to do this with the iris netcdf libraries.  I can check that the attribute is preserved by loading it as a cube, but "CFGroup" doesn't seem to contain accessible information on non-global attributes.

Reply:
Yes, it doesn't appear that Iris has this functionality. It would be good, if we were able to directly check that the `source_realization` is a variable attribute, as that is one of the main points of this work, so it might be worth taking a look at the netCDF4 module (http://unidata.github.io/netcdf4-python/), which might allow you to read the netCDF file more directly.

Reply:
OK that works.  Have updated all three tests to use the netCDF4 module.

Reply:
Great! This looks better as it more directly tests the presence of the variable attributes. Thanks!